From 13c1bbb054a7de55c0f42d1f989776255da2af2f Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Tue, 3 Jul 2018 15:28:29 -0700
Subject: [PATCH 0403/1921] mmc: cavium: Use proper register to clear
 interrupts

The CSR at offset 0x80 (MIO_EMM_INT_W1S) actually sets interrupts instead
of clearing the interrupts. So use MIO_EMM_INT CSR which clears the
interrupts.

Change-Id: I3c65de46260e70840af9420a8903a8774e893aaf
Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
Reviewed-on: https://sj1git1.cavium.com/8531
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-by: Peter Swain <pswain@cavium.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/26898
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/mmc/host/cavium-thunderx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/cavium-thunderx.c b/drivers/mmc/host/cavium-thunderx.c
index 4cfefebf610e..a1c03c8363ac 100644
--- a/drivers/mmc/host/cavium-thunderx.c
+++ b/drivers/mmc/host/cavium-thunderx.c
@@ -116,7 +116,7 @@ static int thunder_mmc_probe(struct pci_dev *pdev,
 	 * Clear out any pending interrupts that may be left over from
 	 * bootloader. Writing 1 to the bits clears them.
 	 */
-	writeq(127, host->base + MIO_EMM_INT_EN(host));
+	writeq(127, host->base + MIO_EMM_INT(host));
 	writeq(3, host->base + MIO_EMM_DMA_INT_ENA_W1C(host));
 	/* Clear DMA FIFO */
 	writeq(BIT_ULL(16), host->base + MIO_EMM_DMA_FIFO_CFG(host));
-- 
2.31.1

